### DISCUSSION TITLE
Problem "pipi numbers"

### DISCUSSION THREAD

**Discussion by GustBatt** [codewars/codewars.com#3190] [Q&A]
My code enters an infinite loop for n >= 6, and I dont know the reason :/

def pipi(n):
    if n == 0:
        return 0
    else:
        ps = [1]
        for i in range(n-1):
            ps += [n]
          
    def check(numb, ind):
    #ind is the nth p, that is, numb
        for i in range(ind):
            if i == 0:
                expression = numb**0.5
            else:
                expression += ps[ind - 1 - i]
                expression = expression**0.5
        if expression == ind:
            return True
        else:
            if expression > ind:
                return 'bigger'
            else:
                return 'lower'
    def numb(numb, exp, ind):
        uBoundNumb = ind**2
        for i in range(ind - 1):
            temp = uBoundNumb
            uBoundNumb = temp**2
        lBoundNumb = (ind)**2 - 2
        for i in range(ind - 1):
            temp = lBoundNumb
            lBoundNumb = temp**2 - 2
        i = 0
        while lBoundNumb != uBoundNumb:
            
            middle = round(((uBoundNumb + lBoundNumb) / 2), 1)
            print(middle)
            i += 1
            if (middle**0.5) % 1 == 0 and check(middle, ind) == True:
                nex = (exp - middle)**2
                return middle, nex
            else:
                if check(middle, ind) == 'bigger':
                    uBoundNumb = middle
                elif check(middle, ind) == 'lower':
                    lBoundNumb = middle
                print(uBoundNumb, lBoundNumb)
    exp = 0
    for i in range(n):
        if i == 0:
            exp = (n**2 - 1)**2
        elif i == n - 2:
            ps[i], ps[i+1] = numb(ps[i], exp, i + 1)
            return ps[i+1]
        else:
            ps[i], exp = numb(ps[i], exp, i + 1)    
    return ps[n-1]

**Comment by hobovsky** [comment]
Codewars Github is meant as a place to discuss functions and issues with Codewars website and platform. It is not meant for help with solving kata. Please ask your question in kata discourse (label your post as `question`), or come for help to `#help-solve` channel of Codewars Discord.

I am closing this discussion as off-topic here. Please ask your question in kata discourse, or on CW Discord.
